apiserver/application/application.go

```diff
@@ -1012,9 +988,108 @@ func (api *API) sameControllerOfferedApplication(url *jujucrossmodel.Application
 
 	// Get the hosting model from the name.
 	sourceModelTag, err = api.sameControllerSourceModel(userName, url.ModelName)
-	if err != nil {
+	if err == nil {
+		app, releaser, err := api.sameControllerOfferedApplication(sourceModelTag, url.ApplicationName, perm)
+		if err != nil {
+			return fail(errors.Trace(err))
+		}
+		defer releaser()
+		offer, err := api.makeOfferParamsFromApplication(url.ApplicationName, app)
+		return offer, sourceModelTag, err
+	}
+	if !errors.IsNotFound(err) {
 		return fail(errors.Trace(err))
 	}
+	return api.differentControllerOfferedApplication(userName, url.ModelName, url.ApplicationName)
+}
+
+func (api *API) differentControllerOfferedApplication(userName, modelName, offerName string) (
+	*params.ApplicationOffer,
+	names.ModelTag,
+	error,
+) {
+	fail := func(err error) (
+		*params.ApplicationOffer,
+		names.ModelTag,
+		error,
+	) {
+		return nil, names.ModelTag{}, err
+	}
+
+	// TODO(wallyworld) - we need a way to pass in the JEM api Info
```

Comment:
I heard you saying JEM yesterday and thought you were saying JUMM in your accent, but obviously not - what's JEM?